### PR TITLE
fix: text Input Field Not Visible on First time direct message

### PR DESCRIPTION
**Pull request by jjinendra3**
## Proposed changes

This PR resolves an issue where the text input field is not visible when initiating a direct message with a user for the first time in the app. The problem stemmed from improper rendering of the text input field, which was caused by the useSubscription hook not functioning correctly. As a result, the placeholder was not being set, leading to errors and ultimately preventing the text input field from being displayed.
## Issue(s)	

#6211 
## How to test or reproduce

1. Go to any thread, chat, or community.
2. Click on the user you have never contacted before.
3. Click on the "Message" button on the user's profile.
4. The text input appears normally as it should 🚀

## Screenshots

https://github.com/user-attachments/assets/60a26f76-5996-454d-8679-b566d7babf8f

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] Improvement (non-breaking change which improves a current function)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [x] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat.ReactNative)
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/containers/MessageComposer/hooks/useSubscription.ts [hunk 1]
```diff
@@ -4,19 +4,30 @@ import { getSubscriptionByRoomId } from '../../../lib/database/services/Subscrip
 import { ISubscription } from '../../../definitions';
 
 // TODO: Not reactive. Should we work on an official version?
+// Please let me know if you'd like me to remove the above comment?
```
Comment (diegolmello) [inline app/containers/MessageComposer/hooks/useSubscription.ts:7]: It's still not reactive. No reason to remove the comment.

File: app/containers/MessageComposer/hooks/useSubscription.ts [hunk 2]
```diff
@@ -4,19 +4,30 @@ import { getSubscriptionByRoomId } from '../../../lib/database/services/Subscrip
...
+
 export const useSubscription = (rid?: string) => {
 	const [subscription, setSubscription] = useState<ISubscription>();
+
 	useEffect(() => {
+		// eslint-disable-next-line no-undef
+		let interval: NodeJS.Timeout;
```
Comment (diegolmello) [inline app/containers/MessageComposer/hooks/useSubscription.ts:14]:
```suggestion
		let interval: ReturnType<typeof setTimeout> | null;
```

File: app/containers/MessageComposer/hooks/useSubscription.ts [hunk 3]
```diff
@@ -4,19 +4,30 @@ import { getSubscriptionByRoomId } from '../../../lib/database/services/Subscrip
...
 			}
 			const result = await getSubscriptionByRoomId(rid);
 			if (result) {
 				setSubscription(result);
 			}
 		};
+
 		loadRoom();
+		if (rid) {
+			interval = setInterval(loadRoom, 50);
```
Comment (diegolmello) [inline app/containers/MessageComposer/hooks/useSubscription.ts:28]: Is this the actual fix or or is it the `return` change on lines 17~18?

**Review by diegolmello — CHANGES_REQUESTED** [review]
Can you write unit tests for this?

File: app/containers/MessageComposer/hooks/useSubscription.ts (hunk 3 above)
Comment (jjinendra3) [reply app/containers/MessageComposer/hooks/useSubscription.ts:28]: This is the fix actually, it works because it prevents multiple overlapping API calls when switching rooms. It ensures that the previous interval is cleared before starting a new one, avoiding unnecessary requests. Additionally, it allows time for the `rid` to be properly set before making the API call, reducing race condition.

**Comment by jjinendra3** [comment]
> Can you write unit tests for this?

Aplogies, didnt see this message earlier, sure, on it!

**Comment by jjinendra3** [comment]
Hey @diegolmello , added some tests for the hook, please let me know if there is an edge case i have missed. Thanks!

**Comment by jjinendra3** [comment]
Hey @diegolmello, I’ve moved the tests into a single file for better modularity. Let me know if I missed anything or if there’s anything that could be improved. Thanks!

**Comment by jjinendra3** [comment]
image

File: app/containers/MessageComposer/hooks/useSubscription.ts (hunk 3 above)
Comment (diegolmello) [reply app/containers/MessageComposer/hooks/useSubscription.ts:26]: Can you elaborate which API calls?
I don't understand why we need such interval.
It looks like there should be a way to do it without having to delay this execution.

Comment (jjinendra3) [reply app/containers/MessageComposer/hooks/useSubscription.ts:26]: Hey @diegolmello ,

The interval is specifically there to handle a race condition with **getSubscriptionByRoomId** API call. When switching to a new direct message, the new room ID (rid) isn’t immediately available. This means that without the interval, we risk calling the API too early—fetching data with an outdated or undefined rid, which leads to an **undefined** subscription.

The 50ms interval helps in two main ways:
- **Ensuring correct data:** It gives the new rid just enough time to be properly set before the API call runs, ensuring that we get the correct subscription data.
- **Preventing overlap:** It makes sure that any previous intervals are cleared before a new one starts, so we avoid overlapping API calls.

Also, this interval only fires a few times during the mounting phase. Once the component is mounted, it **doesn’t** keep firing continuously. It’s not affecting the ongoing chatting or other operations on the page.

I feel this is a pragmatic fix that resolves the timing issue we are facing. Please let me know if you'd like me to explore any additional methods or if there is any approach that you think might yield results in a better way.

Comment (diegolmello) [reply app/containers/MessageComposer/hooks/useSubscription.ts:26]: > The interval is specifically there to handle a race condition with getSubscriptionByRoomId API call. When switching to a new direct message, the new room ID (rid) isn’t immediately available. This means that without the interval, we risk calling the API too early—fetching data with an outdated or undefined rid, which leads to an undefined subscription.

We can check if there's `rid` and don't do anything until we have it.
Add `rid` as dependency.

Comment (jjinendra3) [reply app/containers/MessageComposer/hooks/useSubscription.ts:26]: Hey @diegolmello , 

Apologies for the delay—I was testing various methods to fix the new User DM issue. 

1. **Using useEffect alone:**  
   I tried relying solely on useEffect with `rid` in the dependency array and an early return when `!rid`, but this approach didn’t work.

2. **Using setTimeout with various delays:**  
   I experimented with setTimeout delays (100ms, 500ms, 1000ms) to give `rid` enough time to be set with rid as dependency and `!rid` check. Unfortunately, none of these delays solved the issue.

3. **Current method (using a short interval):**  
   The current solution employs a short-interval timeout (50ms) along with the `!rid` check and `rid` as a dependency (line 33). This method, runs a minimal number of times, is the only one that reliably handles the race condition and ensures that the correct subscription data is fetched. 
   
   

Let me know if you’d like me to explore further alternatives.

Comment (jjinendra3) [reply app/containers/MessageComposer/hooks/useSubscription.ts:26]: Hey @diegolmello,

If there’s an alternative direction you'd like me to explore, I’d be happy to give it a shot. Otherwise, if my initial approach is still not preferable, I’d appreciate any insights on what adjustments could make it more acceptable.

Please let me know your thoughts on the same, thank you.
